### PR TITLE
Create DataBinding Adapter MSD-4

### PR DESCRIPTION
**Pull request by VVMaxVV**
Was created DataBindingRecyclerAdapter which can show all objects that implemented the interface RecyclerItemComparator.
https://trell.ooo/maksymvoloshkointernship/MSD-4

### CODE REVIEW

**Comment by VVMaxVV** [comment]
[Create DataBinding Adapter](https://trello.com/c/mu5SumeM/4-create-databinding-adapter)

File: core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        holder.run {
+            getItem(position).bind(binding)
+            if (binding.hasPendingBindings()) {
+                binding.executePendingBindings()
+            }
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return getItem(position).layoutId
```
Comment (darthoo) [inline core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt:36]: please make this method in body style. `fun()=`

File: core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    fun addAll(itemsList: List<RecyclerItem>) {
```
Comment (darthoo) [inline core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt:39]: This method is redundant. Please remove it

File: core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import com.maxshop.model.RecyclerItem
+
+class DataBindingRecyclerAdapter : ListAdapter<RecyclerItem, BindingViewHolder>(DiffCallback()) {
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): BindingViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding: ViewDataBinding =
+            DataBindingUtil.inflate(inflater, viewType, parent, false)
```
Comment (darthoo) [inline core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt:19]: Please add this line after you create the binding:
`binding.lifecycleOwner = parent.findViewTreeLifecycleOwner()`

**Review by darthoo — CHANGES_REQUESTED** [review]

File: core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt (hunk 1 above)
Comment (VVMaxVV) [reply core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt:36]: fixed

File: core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt (hunk 2 above)
Comment (VVMaxVV) [reply core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt:39]: fixed

File: core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt (hunk 3 above)
Comment (VVMaxVV) [reply core-ui/src/main/java/com/maxshop/adapter/DataBindingRecyclerAdapter.kt:19]: fixed

**Review by darthoo — APPROVED** [review]

**Pull request merged**